daemon/api_validate.go

```diff
@@ -141,7 +144,29 @@ func listValidationSets(c *Command, r *http.Request, _ *auth.UserState) Response
 	return SyncResponse(results, nil)
 }
 
-func getValidationSet(c *Command, r *http.Request, _ *auth.UserState) Response {
+var checkInstalledSnaps = func(vsets *snapasserts.ValidationSets, snaps []*snapasserts.InstalledSnap) error {
+	return vsets.CheckInstalledSnaps(snaps)
+}
+
+func installedSnaps(st *state.State) ([]*snapasserts.InstalledSnap, error) {
+	var snaps []*snapasserts.InstalledSnap
+	all, err := snapstate.All(st)
+	if err != nil {
+		return nil, err
+	}
+	for _, snapState := range all {
+		cur, err := snapState.CurrentInfo()
+		if err != nil {
+			return nil, err
+		}
+		snaps = append(snaps, snapasserts.NewInstalledSnap(
+			snapState.InstanceName(), snapState.CurrentSideInfo().SnapID,
+			cur.Revision))
```

Comment:
nitpicking, indentation is slightly confusing here, feel free to ignore
```suggestion
		snaps = append(snaps,
			snapasserts.NewInstalledSnap(snapState.InstanceName(),
				snapState.CurrentSideInfo().SnapID,
				cur.Revision))
```